gammapy/datasets/spectrum.py

```diff
@@ -58,6 +58,10 @@ def plot_fit(
         >>> dataset.plot_fit(kwargs_residuals=kwargs_residuals, kwargs_spectrum=kwargs_spectrum)  # doctest: +SKIP
         """
         gs = GridSpec(7, 1)
+        if ax_spectrum is None and ax_residuals is None:
+            bool_visible_xticklabel = False
+        else:
+            bool_visible_xticklabel = True
```

Comment:
```suggestion
        bool_visible_xticklabel = not ax_spectrum is None and ax_residuals is None
```

Reply:
Yes but this is missing a bracket I believe:

`bool_visible_xticklabel = not (ax_spectrum is None and ax_residuals is None)`

Reply:
yes you are right

Reply:
Thank you for the suggestion! I modified the code accordingly.